beater/waitready.go

```diff
@@ -79,10 +81,17 @@ func waitReady(
 type waitReadyRoundTripper struct {
 	*http.Transport
 	ready <-chan struct{}
+	drain <-chan struct{}
 }
 
 func (c *waitReadyRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
 	select {
+	case <-c.drain:
+		return &http.Response{
+			Request:    r,
+			StatusCode: http.StatusPreconditionFailed,
+			Body:       io.NopCloser(new(bytes.Buffer)),
+		}, nil
```

Comment:
choosing StatusPreconditionFailed was somewhat arbitrary; it stops the es client from retrying (I originally was returning an error, but then the client would retry X times) and has the word "precondition". Happy to change this to something else.

Reply:
I don't think it matters a great deal, as this is all internal to the server, but I'd prefer not to use `http.StatusPreconditionFailed` as it's a client error (4xx). Perhaps use `http.StatusInternalServerError`?

I think ideally we would return an error. It's possible to customise the retry-on-error behaviour of go-elasticsearch, but it's probably a bit of a rigmarole just for this?